apps/dashboard/src/components/workflow-editor/steps/other-steps-tabs.tsx

```diff
@@ -0,0 +1,143 @@
+/**
+ * This file is used as a placeholder for the other step configuration until the actual configuration is implemented.
+ */
+import { zodResolver } from '@hookform/resolvers/zod';
+import { type StepDataDto, type StepUpdateDto, type WorkflowResponseDto } from '@novu/shared';
+import { Cross2Icon } from '@radix-ui/react-icons';
+import { useForm } from 'react-hook-form';
+import { RiEdit2Line, RiPencilRuler2Line } from 'react-icons/ri';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { Notification5Fill } from '@/components/icons';
+import { Button } from '@/components/primitives/button';
+import { Form } from '@/components/primitives/form/form';
+import { Separator } from '@/components/primitives/separator';
+import { ToastIcon } from '@/components/primitives/sonner';
+import { showToast } from '@/components/primitives/sonner-helpers';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/primitives/tabs';
+import { useUpdateWorkflow } from '@/hooks/use-update-workflow';
+import { buildDefaultValues, buildDynamicZodSchema } from '@/utils/schema';
+import { CustomStepControls } from './controls/custom-step-controls';
+
+const tabsContentClassName = 'h-full w-full px-3 py-3.5';
+
+export const OtherStepTabs = ({ workflow, step }: { workflow: WorkflowResponseDto; step: StepDataDto }) => {
+  const { stepSlug = '' } = useParams<{ workflowSlug: string; stepSlug: string }>();
+  const { dataSchema, uiSchema } = step.controls;
+  const navigate = useNavigate();
+  const schema = buildDynamicZodSchema(dataSchema ?? {});
+  const form = useForm({
+    mode: 'onSubmit',
+    resolver: zodResolver(schema),
+    resetOptions: { keepDirtyValues: true },
+    defaultValues: buildDefaultValues(uiSchema ?? {}),
+    values: step.controls.values,
+  });
+
+  const { reset, formState } = form;
+
+  const { updateWorkflow } = useUpdateWorkflow({
+    onSuccess: () => {
+      showToast({
+        children: () => (
+          <>
+            <ToastIcon variant="success" />
+            <span className="text-sm">Saved</span>
+          </>
+        ),
+        options: {
+          position: 'bottom-right',
+          classNames: {
+            toast: 'ml-10 mb-4',
+          },
+        },
+      });
+    },
+    onError: () => {
+      showToast({
+        children: () => (
+          <>
+            <ToastIcon variant="error" />
+            <span className="text-sm">Failed to save</span>
+          </>
+        ),
+        options: {
+          position: 'bottom-right',
+          classNames: {
+            toast: 'ml-10 mb-4',
+          },
+        },
+      });
+    },
+  });
+
+  const onSubmit = async (data: any) => {
+    await updateWorkflow({
+      id: workflow._id,
+      workflow: {
+        ...workflow,
+        steps: workflow.steps.map((step) =>
+          step.slug === stepSlug ? ({ ...step, controlValues: { ...data } } as StepUpdateDto) : step
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Is it possible to avoid the `as StepUpdateDto` coercion? If not, it's always a good idea to add a small comment to explain why as it's effectively overriding the type system and can result in bugs.‏